### PR TITLE
[AIRFLOW-6170] BranchPythonOperator does not do XCom push of returned value

### PR DESCRIPTION
**Pull request by baolsen**
Make sure you have checked _all_ steps below.

### Jira

- [X] My PR addresses the following [Airflow Jira](https://issues.apache.org/jira/browse/AIRFLOW/) issues and references them in the PR title. For example, "\[AIRFLOW-XXX\] My Airflow PR"
  - https://issues.apache.org/jira/browse/AIRFLOW-XXX
  - In case you are fixing a typo in the documentation you can prepend your commit with \[AIRFLOW-XXX\], code changes always need a Jira issue.
  - In case you are proposing a fundamental code change, you need to create an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)).
  - In case you are adding a dependency, check if the license complies with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).

### Description

- [X] Here are some details about my PR, including screenshots of any UI changes:

Added return_value from BranchPythonOperator along with appropriate test.
### Tests

- [X] My PR adds the following unit tests __OR__ does not need testing for this extremely good reason:

Added a test for XCom push/pull of return_value from BranchPythonOperator
### Commits

- [X] My commits all reference Jira issues in their subject lines, and I have squashed multiple commits if they address the same issue. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
  1. Subject is separated from body by a blank line
  1. Subject is limited to 50 characters (not including Jira issue reference)
  1. Subject does not end with a period
  1. Subject uses the imperative mood ("add", not "adding")
  1. Body wraps at 72 characters
  1. Body explains "what" and "why", not "how"

### Documentation

- [X] In case of new functionality, my PR adds documentation that describes how to use it.
  - All the public functions and the classes in the PR contain docstrings that explain what it does
  - If you implement backwards incompatible changes, please leave a note in the [Updating.md](https://github.com/apache/airflow/blob/master/UPDATING.md) so we can assign it to a appropriate release

N/A

### CODE REVIEW

**Review by potiuk — APPROVED** [review]
Nice catch @baolsen !

**Comment by potiuk** [comment]
One small request here @baolsen -> this is not a breaking change, so this is not a MUST, but it would be great to make a note about this change in behaviour of BranchPythonOperator in UPDATING.md. Would you please make a fixup and add this info?

File: UPDATING.md [hunk 1]
```diff
@@ -41,6 +41,11 @@ assists users migrating to a new version.
 
 ## Airflow Master
 
+### BranchPythonOperator has a return value
```
Comment (potiuk) [inline UPDATING.md:50]: ❤️

**Comment by baolsen** [comment]
Hey @potiuk , some flaky tests again - please rerun when you can.
FYI I've noticed that its often the Cassandra Hook tests that fail, perhaps that service is not started up in time for the test

**Comment by potiuk** [comment]
Yep @baolsen restarted. And yes we know cassandra is a problem sometimes - likely some resource race on Travis machines. however hopefully soon we move to Github Actions (as of recently they support custom runners and we can switch to our own GCP runners - we have some credits from Google) and we can optimise this /improve stability there @nuclearpinguin has done POC on github actions and they seem to work well.

**Comment by baolsen** [comment]
Fixed conflicts (UPDATING.md)

File: UPDATING.md [hunk 2]
```diff
@@ -46,6 +46,10 @@ assists users migrating to a new version.
 This option has been removed because it is no longer supported by the Google Kubernetes Engine. The new
 recommended service account keys for the Google Cloud Platform management method is
 [Workload Identity](https://cloud.google.com/kubernetes-engine/docs/how-to/workload-identity).
+### BranchPythonOperator has a return value
+`BranchPythonOperator` will now return a value equal to the `task_id` of the chosen branch, 
+where previously it returned None. Since it inherets from BaseOperator it will do an
```
Comment (feluelle) [inline UPDATING.md]:
```suggestion
where previously it returned None. Since it inherits from BaseOperator it will do an 
```

File: UPDATING.md [hunk 3]
```diff
@@ -46,6 +46,10 @@ assists users migrating to a new version.
 This option has been removed because it is no longer supported by the Google Kubernetes Engine. The new
 recommended service account keys for the Google Cloud Platform management method is
 [Workload Identity](https://cloud.google.com/kubernetes-engine/docs/how-to/workload-identity).
+### BranchPythonOperator has a return value
```
Comment (feluelle) [inline UPDATING.md:50]:
```suggestion

### BranchPythonOperator has a return value
```

**Review by feluelle — COMMENTED** [review]
LGTM overall 👍

File: UPDATING.md (hunk 2 above)
Comment (baolsen) [reply UPDATING.md]: Thanks, I've fixed up both of these

**Comment by Fokko** [comment]
Restarted the build @baolsen

**Comment by baolsen** [comment]
Build is failing due to flaky tests but I have requested review meanwhile anyway.

**Review by Fokko — APPROVED** [review]
I'm surprised by the fact that the `BranchPythonOperator` did not return anything.

**Review by potiuk — APPROVED** [review]

**Comment by potiuk** [comment]
Thanks @baolsen ! It was indeed an overlook!

**Pull request merged**